gensim/models/ensemblelda.py

```diff
@@ -0,0 +1,1308 @@
+
+                processes.append(process)
+                pipes.append((parent_conn, child_conn))
+                process.start()
+
+            except ProcessError:
+                self._teardown(pipes, processes, i)
+                raise ProcessError
+
+        distances = []
+        # note, that the following loop maintains order in how the ttda will be concatenated
+        # which is very important. Ordering in ttda has to be the same as when using only one process
+        for parent_conn, _ in pipes:
+            answer = parent_conn.recv()
+            parent_conn.close()  # child conn will be closed from inside the worker
+            # this does basically the same as the _generate_topic_models function (concatenate all the ttdas):
+            distances.append(answer[1])
+
+        # end all processes
+        for process in processes:
+            process.terminate()
+
+        self.asymmetric_distance_matrix = np.concatenate(distances)
+
+        return self.asymmetric_distance_matrix
+
+    def _calculate_asymmetric_distance_matrix_chunk(self, ttda1, ttda2, threshold, start_index, method):
+        """Calculate an (asymmetric) distance from each topic in ``ttda1`` to each topic in ``ttda2``.
+
+        Parameters
+        ----------
+        ttda1 and ttda2: 2D arrays of floats
+            Two ttda matrices that are going to be used for distance calculation. Each row in ttda corresponds to one
+            topic. Each cell in the resulting matrix corresponds to the distance between a topic pair.
+        threshold : float, optional
+            threshold defaults to: ``{"mass": 0.95, "rank": 0.11}``, depending on the selected method
+        start_index : int
+            this function might be used in multiprocessing, so start_index has to be set as ttda1 is a chunk of the
+            complete ttda in that case. start_index would be 0 if ``ttda1 == self.ttda``. When self.ttda is split into
+            two pieces, each 100 ttdas long, then start_index should be be 100. default is 0
+        method : {'mass', 'rank}, optional
+            method can be "mass" for the original masking method or "rank" for a faster masking method that selects
+            by rank of largest elements in the topic term distribution, to determine which tokens are relevant for the
+            topic.
+
+        Returns
+        -------
+        2D numpy.ndarray of floats
+            Asymmetric distance matrix of size ``len(ttda1)`` by ``len(ttda2)``.
+
+        """
+        # initialize the distance matrix. ndarray is faster than zeros
+        distances = np.ndarray((len(ttda1), len(ttda2)))
+
+        if ttda1.shape[0] > 0 and ttda2.shape[0] > 0:
+            # the worker might not have received a ttda because it was chunked up too much
+
+            if method not in ["mass", "rank"]:
+                raise ValueError(f"method {method} unknown")
+
+            # select masking method:
+            def mass_masking(a):
```

Comment:
The benefit of keeping these nested is that you do not need to pass threshold is a parameter. Are there any others?

The cost of nesting these functions is that you can no longer refer to them by directl (you have to pass strings, do a dictionary lookup) and thus you cannot test them directly. 

Do the benefits justify the cost? In my opinion, in this case the answer is "no".